### ISSUE TITLE
Stat weights do not work on rogue

### ISSUE THREAD

**Issue by PaulHalliday** [wowsims/cata#578]
**Issue**

When attempting to determine stat weights for this [Assassination rogue](https://wowsims.github.io/cata/rogue/assassination/)
```
{"talents":"2330230113322110321-1030000000000000000-0010000000000000000","glyphs":{"minor":[{"spellID":58038},{"spellID":58032},{"spellID":58033}],"prime":[{"spellID":63268},{"spellID":56800},{"spellID":56801}],"major":[{"spellID":56811},{"spellID":63254}]},"class":"rogue","unit":"player","race":"Human","name":"Darksorrow","gear":{"items":[{"id":55669,"gems":[],"random_suffix":-91},{"id":55804,"gems":[]},{"id":55204,"gems":[]},{"id":57359,"gems":[]},{"id":55642,"gems":[],"random_suffix":-91},{"id":55633,"gems":[],"random_suffix":-136},{"id":55660,"gems":[],"random_suffix":-91},{"id":55696,"gems":[],"random_suffix":-136},{"id":55678,"gems":[],"random_suffix":-135},{"id":57279,"gems":[]},{"id":55776,"gems":[]},{"id":52310,"gems":[],"random_suffix":-136},{"id":57325,"gems":[]},{"id":55266,"gems":[]},{"id":55044,"gems":[]},{"id":55044,"gems":[]},{"id":57347,"gems":[]}]},"professions":[{"name":"Engineering","level":360},{"name":"Jewelcrafting","level":151}],"level":82,"spec":"assassination","realm":"Firemaw"}
```

It fails after 2 iterations with the following in the JS console:

```
Uncaught (in promise) RuntimeError: null function or function signature mismatch
    at github.com_wowsims_cata_sim_core.__APLRotation_.getNextAction (lib.wasm:0x63d7ce)
    at github.com_wowsims_cata_sim_core.__APLRotation_.DoNextAction (lib.wasm:0x63d190)
    at github.com_wowsims_cata_sim_core.__Environment_.initialize.__Character_.initialize.func1 (lib.wasm:0x70c309)
    at github.com_wowsims_cata_sim_core.__Simulation_.Step (lib.wasm:0x770593)
    at github.com_wowsims_cata_sim_core.__Simulation_.runOnce (lib.wasm:0x76e155)
    at github.com_wowsims_cata_sim_core.__Simulation_.run (lib.wasm:0x76ce71)
    at wasm_pc_f_loop (lib.wasm:0x13559e)
    at wasm_export_resume (lib.wasm:0x13566d)
    at globalThis.Go._resume (sim_worker.js:578:26)
    at sim_worker.js:588:13
```

**Comment by Tharre** [comment]
Can't reproduce, but I've seen this happen before on some conditionals.

Try resetting your settings to default, and try again and see if the error persists.

**Comment by MichaelRoehrig** [comment]
Steps to reproduce:

1. Set everything to default
2. Load the export data below.
3. Set duration to 300, iterations to 1,000,000.
4. Calculate Stat Weights.

Result: Nothing happens, server console doesn't show anything.

Error: Promised response from onMessage listener went out of scope [ExtensionMessagingService.js:89:34](moz-extension://918dc923-0e74-4d44-9681-ce3a470a1965/scripts/services/ExtensionMessagingService.js)

`{"talents":"0333230013122110321-0020000000000000000-2030030000000000000","glyphs":{"minor":[{"spellID":58038},{"spellID":58039},{"spellID":58032}],"prime":[{"spellID":63249},{"spellID":63268},{"spellID":56801}],"major":[{"spellID":63254},{"spellID":56804},{"spellID":56811}]},"class":"rogue","unit":"player","race":"Troll","name":"Bonwart","gear":{"items":[{"id":59455,"gems":[68778,59489,59493],"enchant":4209},{"id":69827,"gems":[],"reforging":153,"random_suffix":-136},{"id":56273,"gems":[52212],"reforging":154,"enchant":4204},{"id":62361,"gems":[],"reforging":154,"enchant":1099},{"id":65060,"gems":[52212,52212],"reforging":146,"enchant":4063},{"id":56409,"gems":[],"reforging":154,"enchant":4089},{"id":62417,"gems":[52212],"reforging":147,"enchant":4061},{"id":56537,"gems":[52212,52212]},{"id":60300,"gems":[52204,52220],"reforging":168,"enchant":4126},{"id":58482,"gems":[52220],"reforging":147,"enchant":4076},{"id":62362,"gems":[],"reforging":153,"enchant":4079},{"id":67136,"gems":[],"reforging":154,"enchant":4079},{"id":58181,"gems":[]},{"id":62463,"gems":[],"reforging":168},{"id":56302,"gems":[],"reforging":168,"enchant":4083},{"id":56335,"gems":[],"reforging":151,"enchant":4083},{"id":68608,"gems":[],"reforging":151}]},"professions":[{"name":"Engineering","level":525},{"name":"Enchanting","level":525}],"level":85,"spec":"assassination","realm":"Venoxis"}`

**Comment by rosenrusinov** [comment]
@PaulHalliday You are trying to sim with a character that is not max level and we dont support this so errors are expected.

@MichaelRoehrig I tried reproducing with your import string and I cant. Ran a 50,000 iterations stat weights with no issues. When you are running stat weights with so many iterations it is expected for it to take so much time. 50k iterations on the rogue stat weights ends up at 0.6 million iterations which is quite a lot of computing to do.
